Mathlib/Computability/NFA.lean

```diff
@@ -1,20 +1,41 @@
 /-
 Copyright (c) 2020 Fox Thomson. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Fox Thomson
+Authors: Fox Thomson, Maja Kądziołka
 -/
 import Mathlib.Computability.DFA
 import Mathlib.Data.Fintype.Powerset
 
 /-!
 # Nondeterministic Finite Automata
-This file contains the definition of a Nondeterministic Finite Automaton (NFA), a state machine
-which determines whether a string (implemented as a list over an arbitrary alphabet) is in a regular
-set by evaluating the string over every possible path.
-We show that DFA's are equivalent to NFA's however the construction from NFA to DFA uses an
-exponential number of states.
-Note that this definition allows for Automaton with infinite states; a `Fintype` instance must be
-supplied for true NFA's.
+
+A Nondeterministic Finite Automaton (NFA) is a state machine which
+decides membership in a particular `Language`, by following every
+possible path that describes an input string.
+
+We show that DFAs and NFAs can decide the same languages, by constructing
+an equivalent DFA for every NFA, and vice versa.
+
+As constructing a DFA from an NFA uses an exponential number of states,
+we re-prove the pumping lemma instead of lifting `DFA.pumping_lemma`,
+in order to obtain the optimal bound on the minimal length of the string.
+
+Like `DFA`, this definition allows for automata with infinite states;
+a `Fintype` instance must be supplied for true NFAs.
+
+## Main definitions
+
+* `NFA α σ`: automaton over alphabet `α` and set of states `σ`
+* `M.evalFrom S x`: set of possible ending states for an input word `x`
+  and set of initial states `S`
+* `M.accepts`: the language accepted by the NFA `M`
+* `M.Path s t x`: a specific path from `s` to `t` for an input word `x`
+* `p.supp`: set of states visited by the path `p`
```

Comment:
```suggestion
* `NFA.evalFrom M S x`: set of possible ending states for an input word `x`
  and set of initial states `S`
* `NFA.accepts M`: the language accepted by the NFA `M`
* `NFA.Path M s t x`: a specific path from `s` to `t` for an input word `x`
* `NFA.Path.supp p`: set of states visited by the path `p`
```
so that doc-gen turns them into links